### PR TITLE
Shane/autofill username poc

### PR DESCRIPTION
**Pull request by shakyShane**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/1202427674957632/1202534920365235/f
iOS PR:  https://github.com/duckduckgo/iOS/pull/1225
macOS PR: https://github.com/more-duckduckgo-org/macos-browser/pull/646
What kind of version bump will this require?: Patch

**Description**:

Please see the Asana task, but the short version is that we want to prevent trying to Autofill a password-only credential when a user has focussed into the username field. 

**Steps to test this PR**:

**Testing `EMPTY` username**

1. Save or Create a login with an empty username (I use amazon.com login form for this)
1. Sign out, and then when trying to log back in, ensure the native modal is **not** shown when you focus the username field

**Testing `non-EMPTY` username**

1. Save or Create a login with a username + password
1. Sign out, and then when trying to log back in, ensure the native modal **is** shown when you focus the username field

**OS Testing**:

* [ ] iOS 13
* [ ] iOS 14
* [ ] iOS 15
* [ ] macOS 10.15
* [ ] macOS 11

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/BrowserServicesKit/Autofill/AutofillUserScript+SecureVault.swift [hunk 1]
```diff
@@ -322,14 +339,14 @@ extension AutofillUserScript {
...
+        guard let req: GetAutofillDataRequest = decodeMessageBody(from: message.messageBody) else {
+            return;
         }
-        if mainType == "credentials" {
+
+        if req.mainType == .credentials {
             let domain = hostForMessage(message)
             
-            vaultDelegate?.autofillUserScript(self, didRequestCredentialsForDomain: domain) { credentials, action in
+            vaultDelegate?.autofillUserScript(self, didRequestCredentialsForDomain: domain, subType: req.subType) { [self] credentials, action in
```
Comment (Bunn) [inline Sources/BrowserServicesKit/Autofill/AutofillUserScript+SecureVault.swift:349]: Any reason to retain self here?

**Review by Bunn — COMMENTED** [review]
Couple comments, but the overall approach LGTM

File: Sources/BrowserServicesKit/Autofill/AutofillUserScript+SecureVault.swift (hunk 1 above)
Comment (shakyShane) [reply Sources/BrowserServicesKit/Autofill/AutofillUserScript+SecureVault.swift:349]: oops! left behind by mistake, removing.

**Review by Bunn — APPROVED** [review]
There are some GH action linter warnings that needs to be addressed, but otherwise LGTM

**Pull request merged**